src/main/java/com/amazon/opendistroforelasticsearch/sql/metrics/NumericMetric.java

```diff
@@ -0,0 +1,51 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.metrics;
+
+public class NumericMetric<T> extends Metric<T> {
+
+    private Counter<T> counter;
+
+    public NumericMetric(String name, Counter counter) {
```

Comment:
I think it's not a good idea to "outsource" a concrete type's behavior. What's the use case where we would need to change how we count? And the second comment, even if you want to support this, it should be an override to the default, in other words there should be a constructor which accepts just the name and uses `BasicCounter` as the `Counter` implementation.

Reply:
Ok, never mind, I guess this is fine. However, I am thinking if it would be better to derive the Counter implementation from the `MetricType`, I got the impression that those are somehow connected. Anyway, this is an `NP` comment, just out of my curiousity.

Reply:
Thanks.